tests/utils.go

```diff
@@ -86,13 +86,11 @@ func GetRunningVMIDomainSpec(vmi *v1.VirtualMachineInstance) (*launcherApi.Domai
 
 func CheckCloudInitMetaData(vmi *v1.VirtualMachineInstance, testFile, testData string) {
 	cmdCheck := "cat " + filepath.Join("/mnt", testFile) + "\n"
-	res, err := console.SafeExpectBatchWithResponse(vmi, []expect.Batcher{
+	err := console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "sudo su -\n"},
 		&expect.BExp{R: console.PromptExpression},
 		&expect.BSnd{S: cmdCheck},
 		&expect.BExp{R: testData},
 	}, 15)
-	if err != nil {
```

Comment:
Maybe the problem is that the "check" is too odd to interpret correctly.
It seems that:
- If the content of the test file is equivalent to the test data, the check passes.
- If the content of the file is not as expected but nevertheless the response did include the test data as a subset, it will also be considered as passed.

So I guess the new simplification is wrong.

You should probably just return the result and error, letting the caller assert the expectation.
But this is really confusing and hard to interpret correctly,